src/hotspot/cpu/x86/x86.ad

```diff
@@ -5775,6 +5775,46 @@ instruct evminmaxFP_reg_eavx(vec dst, vec a, vec b, vec atmp, vec btmp, kReg ktm
   ins_pipe( pipe_slow );
 %}
 
+// --------------------------------- Signum ---------------------------
+
+instruct signumF_reg(regF dst, regF zero, regF one, rFlagsReg cr) %{
+  match(Set dst (SignumF dst (Binary zero one)));
+  effect(KILL cr);
+  format %{ "signumF $dst, $dst" %}
+  ins_encode %{
+    Label exit;
+
+    __ ucomiss($dst$$XMMRegister, $zero$$XMMRegister);
+    __ jcc(Assembler::parity, exit);
```

Comment:
How about checking equal first and then parity?

I think the unordered case is rare in real programs.

Reply:
+1